hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -386,6 +386,11 @@ public void start() {
             startHeartbeat();
             startIcmpPing();
             connectionManager.connectToCluster();
+            ClientConnectionStrategyConfig connectionStrategyConfig = config.getConnectionStrategyConfig();
+            boolean asyncStart = connectionStrategyConfig.isAsyncStart();
+            if (!asyncStart) {
+                sendStateToCluster();
```

Comment:
just a question: any reason why we have moved this call for non-async client to here, instead of at the end?

Reply:
No there is not. Let me move it to end.